packages/neon-core/src/sc/NEF.ts

```diff
@@ -32,6 +32,10 @@ export class NEF {
     this.tokens = tokens.map((token) => new sc.MethodToken(token));
     this.script = script;
     this.checksum = checksum;
+    // this avoids having to compute and update the checksum after the object is created
+    if (this.checksum == 0) {
+      this.checksum = this.computeCheckSum();
+    }
```

Comment:
this prevents the C# way of creating a NEF, which is something like
```C#
var nef = new NefFile
            {
                Compiler = "test-compiler 0.1",
                Script = new byte[] {(byte) OpCode.RET},
                Tokens = new MethodToken[0],
            };
            nef.CheckSum = NefFile.ComputeChecksum(nef);
```

Reply:
Yes correct, checksums are for integrity. Checksum of zero is not possible but that's beside the point. We need a value in the `checksum` field in order to be able to compute the checksum because the computation first serializes the object and then drops the last 4 bytes of the "default" checksum before hashing it and creating the real checksum. 

Perhaps we should just drop `checksum` from the `NEFLike` interface and always compute it?

Reply:
Yes, drop it in the constructor. We can just do the check when importing it from an external source. We can put it behind a getter. Have a prrivate method for serializating without checksum so it is clearer and we dont have to depend on a "default" value.

Reply:
> We can put it behind a getter.
Not sure what you mean by this, but have a look at how I solved it